tests/parse_examples.rs

```diff
@@ -1,81 +1,27 @@
-use factorio_blueprint::{BlueprintCodec, Container};
-use std::io::{BufReader, Read};
-use std::path::PathBuf;
-
-fn examples() -> impl Iterator<Item = PathBuf> {
-    let mut examples = std::env::current_exe().expect("can find test executable");
-    // target/debug/1/executable
-    for _ in 0..4 {
-        examples.pop();
-    }
-    examples.push("tests/examples");
-
-    std::fs::read_dir(examples)
-        .expect("should find examples dir")
-        .map(|maybe_example| maybe_example.expect("should find file").path())
-        .inspect(|example| {
-            dbg!(example);
-        })
-}
-
-fn container_examples() -> impl Iterator<Item = Container> {
-    examples().map(|path| {
-        BlueprintCodec::decode(BufReader::new(
-            std::fs::File::open(path).expect("can open example file"),
-        ))
-        .expect("container decoded successfully")
-    })
-}
-
-fn test_parse<R: Read>(reader: R) {
-    let mut json_data = Vec::new();
-    BlueprintCodec::decode_reader(reader, |mut reader| {
-        reader.read_to_end(&mut json_data).map(|_| ())
-    })
-    .expect("should have expected version");
-    let json_str = String::from_utf8(json_data).expect("json data should be valid utf-8");
-
-    if let Err(jserr) = serde_json::from_str::<Container>(&json_str) {
-        use serde_json::error::Category;
-        match jserr.classify() {
-            Category::Syntax | Category::Data => {
-                const SPACING: usize = 50;
-                let mut col = jserr.column();
-                col -= col.min(SPACING);
-                println!("bad json: {}", &json_str[col..col + 50]);
-            }
-            _ => {}
+#[cfg(test)]
+mod tests {
+    use std::path::PathBuf;
+
+    fn examples() -> impl Iterator<Item = PathBuf> {
+        let mut examples = std::env::current_exe().expect("can find test executable");
+        // target/debug/1/executable
+        for _ in 0..4 {
+            examples.pop();
         }
-        panic!("error: {}", jserr);
+        examples.push("tests/examples");
+
+        std::fs::read_dir(examples)
+            .expect("should find examples dir")
+            .map(|maybe_example| maybe_example.expect("should find file").path())
+            .inspect(|example| {
+                dbg!(example);
+            })
     }
-}
 
-#[test]
-fn can_parse_examples() {
-    for example in examples() {
-        let file_data = std::fs::read_to_string(example).expect("can read from file");
-
-        test_parse(file_data.trim().as_bytes());
-    }
-}
-
-#[test]
-fn can_read_from_untrimmed_files() {
```

Comment:
Why remove these tests?

Reply:
Seemed a bit overkill to read/parse the files 3x when the roundtrip is basically a superset of the other two.

Reply:
Roundtrip is not a superset of the other two. In particular, there's a category of bug in which the parser is sensitive to whitespace. That gets revealed if `can_parse_examples` xor `can_read_from_untrimmed_files` passes. Please restore these tests.